django/db/backends/sqlite3/schema.py

```diff
@@ -95,7 +95,9 @@ def is_self_referential(f):
         # If any of the new or altered fields is introducing a new PK,
         # remove the old one
         restore_pk_field = None
-        if any(f.primary_key for f in create_fields) or any(n.primary_key for o, n in alter_fields):
+        if any([
+                getattr(create_field, 'primary_key', False),
```

Comment:
No need to use `any`... instead: `getattr(create_field, 'primary_key', False) or (a and b) ...`